ksqldb-engine/src/main/java/io/confluent/ksql/function/UdafFactoryInvoker.java

```diff
@@ -70,10 +72,12 @@
     if (!Modifier.isStatic(method.getModifiers())) {
       throw new KsqlException("UDAF factory methods must be static " + method);
     }
-    final UdafTypes types = new UdafTypes(method, functionName, typeParser);
+    this.types = new UdafTypes(method, functionName, typeParser);
     this.functionName = Objects.requireNonNull(functionName);
-    this.aggregateArgType = Objects.requireNonNull(types.getAggregateSchema(aggregateSchema));
-    this.aggregateReturnType = Objects.requireNonNull(types.getOutputSchema(outputSchema));
+    this.aggregateSchema = aggregateSchema;
+    this.outputSchema = outputSchema;
```

Comment:
nit: we should keep the `Objects.requireNonNull` (unless it can be null?) also remove the commented out lines :)

Reply:
This value will be null if the annotation is unused.